lib/index.js

```diff
@@ -130,4 +133,87 @@ Dynamoose.prototype.revertDDB = function () {
   this.dynamoDB = null;
 };
 
+Dynamoose.prototype.transaction = async function(items, options, next) {
+  debugTransaction('Run Transaction');
+  const deferred = Q.defer();
+
+  options = options || {};
+  if(typeof options === 'function') {
+    next = options;
+    options = {};
+  }
+
+  if(items === null || items === undefined || items.length === 0) {
+    deferred.reject(new errors.TransactionError('Items required to run transaction'));
+    return deferred.promise.nodeify(next);
+  }
+
+  items = await Promise.all(items);
+  let transactionReq = {
+    TransactItems: items.map(item => {
+      const returnItem = {...item};
+      delete returnItem.$__;
+      return returnItem;
+    })
+  };
+  let transactionMethodName = items.map(obj => Object.keys(obj)[0]).every(key => key === "Get") ? "transactGetItems" : "transactWriteItems";
+  if (options.type) {
+    debugTransaction("Using custom transaction method");
+    if (options.type === "get") {
+      transactionMethodName = "transactGetItems";
+    } else if (options.type === "write") {
+      transactionMethodName = "transactWriteItems";
+    } else {
+      deferred.reject(new errors.TransactionError('Invalid type option, please pass in "get" or "write"'));
+      return deferred.promise.nodeify(next);
+    }
+  } else {
+    debugTransaction("Using predetermined transaction method");
+  }
+  debugTransaction(`Using transaction method: ${transactionMethodName}`);
+
+  const transact = () => {
+    console.log(JSON.stringify(transactionReq));
```

Comment:
lets not forget to 🔥 this in final implementation.

Reply:
@jutaz Of course. Will do that. Thanks for the reminder.

Reply:
Complete in latest commit